src/feature/hs/hs_common.c

```diff
@@ -1698,6 +1698,14 @@ hs_get_extend_info_from_lspecs(const smartlist_t *lspecs,
 
   SMARTLIST_FOREACH_BEGIN(lspecs, const link_specifier_t *, ls) {
     switch (link_specifier_get_ls_type(ls)) {
+    case LS_IPV4:
+      /* Skip if we already seen a v4. */
```

Comment:
Please update the comment to say why we skip when direct_conn is true.

Reply:
Please also update the comment to say what we're doing here:

"If direct_conn is false, initialise ap with the first IPv4 address and port in the link specifiers."